theano/gpuarray/extra_ops.py

```diff
@@ -48,9 +49,8 @@ def c_header_dirs(self):
         return [os.path.dirname(__file__)]
 
     def get_params(self, node):
-        context = GpuKernelBase.get_params(self, node)
-        device_is_not_cuda = (context.kind != b'cuda')
-        return self.params_type.get_params(axis=self.axis, context=context, device_is_not_cuda=device_is_not_cuda)
+        context = node.inputs[0].type.context
```

Comment:
I see that the inputs[0] restriction is needed here. Why not keep the old line 51 and remove that restriction?